src/bp/ui-admin/src/Pages/Confusion/index.jsx

```diff
@@ -0,0 +1,139 @@
+import React, { Component } from 'react'
+
+import SectionLayout from '../Layouts/Section'
+import Details from './details'
+
+import _ from 'lodash'
+
+import api from '../../api'
+
+const allFilterFields = ['lang', 'version']
+
+class Confusion extends Component {
+  state = {
+    confusions: [],
+    botIds: [],
+    select: ['', ''],
+    isComputing: false,
+    version: ''
+  }
+
+  componentDidMount = async () => {
+    this.addConfusionToState(await this.getAllConfusions())
+    this.initLabels()
+  }
+
+  versionChange = event => this.setState({ version: event.target.value })
+
+  setSelect = (value, index) =>
+    this.setState({ select: [...this.state.select.slice(0, index), value, ...this.state.select.slice(index + 1)] })
+
+  selectChangeFromFrontEnd = index => event => this.setSelect(event.target.value, index)
+
+  getAllBotIds = async () => {
+    if (_.isEmpty(this.state.botIds)) {
+      this.setState({ botIds: (await api.getSecured().get('/admin/bots')).data.payload.bots.map(bot => bot.id) })
+    }
+
+    return this.state.botIds
+  }
```

Comment:
Bot IDS should be fetched from the store, in the component mount. No need for the api call